src/make_watertight/Cleanup.cpp

```diff
@@ -225,13 +219,13 @@ moab::ErrorCode Cleanup::delete_small_edges(const moab::Range& surfaces, const d
       }
       assert(3 <= adj_edges1.size());
       moab::Range adj_skin_edges1 = intersect(adj_edges1, skin_edges);
-      bool endpt1_is_skin;
-      if (adj_skin_edges1.empty())
-        endpt1_is_skin = false;
-      else
-        endpt1_is_skin = true;
-      if (endpt0_is_skin && endpt1_is_skin)
+
+      // check to see if the endpoints part of the skin
+      bool endpt0_is_skin = !adj_skin_edges0.empty();
+      bool endpt1_is_skin = !adj_skin_edges1.empty();
+      if (endpt0_is_skin && endpt1_is_skin) {
```

Comment:
```suggestion
      if (!adj_skin_edges1.empty() && !adj_skin_edges0.empty()) {
```

Reply:
Had this thought as well... I left those variables in there for clarity though. The fact that the `Range` `adj_skin_edges1` is empty doesn't quite get across the point that the endpoint of the edges is a skin point like the Boolean variable names do. I can update it if you're feeling strongly about it though.

Reply:
I don't have a string feeling either....